backend/comm.c

```diff
@@ -506,7 +506,9 @@ remote_server * get_remote_server(char *hostname, unsigned short portno,
 
         if(do_connect)
         {
-			rs->sockfd = socket(AF_INET, SOCK_STREAM, 0);
+			rs->status = NODE_LIVE;
```

Comment:
I don't understand why you are bringing this back. I moved setting `rs->status = NODE_LIVE` down to after the connect, when we actually know if the connect succeeded or not. Meanwhile, `rs->status = NODE_UNKNOWN` as set on line 481. Isn't that more correct? We don't actually know if the server is live until after we connected, thus NODE_UNKNOWN is the status we should use?

Reply:
Okay. I figured that as the comms is one thread and the RTS threads are running concurrently, we can access things in parallel and see these transitions as they happen and thus, marking the NODE_LIVE too early would have the RTS threads attempt to send data to this server (which isn't connected yet). But I don't have the full workings of this in my head so I'm probably have a mistake in my reasoning somewhere...

Reply:
The concurrency concern is valid, but by the way we update the local view in the client API, the active list of servers never gets updated with a new remote_server pointer until we have settled on whether the server is NODE_DEAD or NODE_LIVE, so we would never see a transitory state. Also, concurrent updates to that list are protected by db->gossip_lock (although at the moment updates only happen from the comm thread, so that's redundant at the moment).
The initial idea of the NODE_UNKNOWN state was not to handle access concurrency, but to support an intermediate state in inter-server gossip where nodes are learned about, but their liveness hasn't yet been verified by connection. However, I realized in practice that state is superfluous, NODE_DEAD will do just fine for that.